src/include/cursor.i

```diff
@@ -402,15 +402,17 @@ __cursor_func_init(WT_CURSOR_BTREE *cbt, bool reenter)
 
     /* Activate the file cursor. */
     if (!F_ISSET(cbt, WT_CBT_ACTIVE)) {
-        if (!F_ISSET(cbt, WT_CBT_NO_TRACKING))
+        if (!F_ISSET(cbt, WT_CBT_NO_TXN))
             WT_RET(__cursor_enter(session));
         F_SET(cbt, WT_CBT_ACTIVE);
     }
 
     /*
      * If this is an ordinary transactional cursor, make sure we are set up to read.
      */
-    if (!F_ISSET(cbt, WT_CBT_NO_TXN))
+    if (F_ISSET(cbt, WT_CBT_HS))
```

Comment:
If it's a history store cursor, explicitly pin ids since we're not protected from obsolete checks.

This was previously the only shared functionality between non-transactional checkpoint cursors and history store ones. Now that they don't share anything, I've renamed the checkpoint specific flag back to `WT_CBT_NO_TXN` and have put the history store specific logic behind `WT_CBT_HS`.